darts/models/theta.py

```diff
@@ -115,14 +120,241 @@ def predict(self, n: int) -> 'TimeSeries':
+                 trend_mode: Trend = Trend.LINEAR,
+                 normalization: bool = True):
+        """
+        An implementation of the 4Theta method with configurable `theta` parameter.
+
+        See M4 competition `solution <https://github.com/Mcompetitions/M4-methods/blob/master/4Theta%20method.R>`_.
+
+        The training time series is de-seasonalized according to `seasonality_period`,
+        or an inferred seasonality period.
+
+        When called with `theta = 2 - X`, `model_mode = Model.ADDITIVE` and `trend_mode = Trend.LINEAR`,
+        this model is equivalent to calling `Theta(theta=X)`.
+
+        Parameters
+        ----------
+        theta
+            Value of the theta parameter. Defaults to 2.
+            If theta = 1, then the fourtheta method restricts to a simple exponential smoothing (SES).
+            If theta = 0, then the fourtheta method restricts to a simple `trend_mode` regression.
+        seasonality_period
+            User-defined seasonality period. If not set, will be tentatively inferred from the training series upon
+            calling `fit()`.
+        model_mode
+            Type of model combining the Theta lines. Must be a Model Enum member. Defaults to `Model.ADDITIVE`.
+        season_mode
+            Type of seasonality. Must be a Season Enum member. Defaults to `Season.MULTIPLICATIVE`.
+        trend_mode
+            Type of trend to fit. Must be a Trend Enum member. Defaults to `Trend.LINEAR`.
+        normalization
+            If `True`, the data is normalized so that the mean is 1. Defaults to `True`.
+        """
+
+        super().__init__()
+
+        self.model = None
+        self.drift = None
+        self.mean = 1
+        self.length = 0
+        self.theta = theta
+        self.is_seasonal = False
+        self.seasonality = None
+        self.season_period = seasonality_period
+        self.model_mode = model_mode
+        self.season_mode = season_mode
+        self.trend_mode = trend_mode
+        self.wses = 0 if self.theta == 0 else (1 / theta)
+        self.wdrift = 1 - self.wses
+        self.fitted_values = None
+        self.normalization = normalization
+
+        raise_if_not(model_mode in Model,
+                     "Unknown value for model_mode: {}.".format(model_mode), logger)
+        raise_if_not(trend_mode in Trend,
+                     "Unknown value for trend_mode: {}.".format(trend_mode), logger)
+        raise_if_not(season_mode in Season,
+                     "Unknown value for season_mode: {}.".format(season_mode), logger)
+
+    def fit(self, ts, component_index: Optional[int] = None):
+        super().fit(ts, component_index)
+
+        self.length = len(ts)
+        # normalization of data
+        if self.normalization:
+            self.mean = ts.mean().mean()
+            raise_if_not(not np.isclose(self.mean, 0),
+                         "The mean value of the TimeSeries must not be 0 when using normalization", logger)
```

Comment:
Perhaps you can write "The mean value of the provided series is too close to zero to perform normalization"